File: agrimonitor_backend/app/schemas/recommendation.py
from pydantic import BaseModel


class RecommendationMatch(BaseModel):
    disease_rule_id: int
    disease_name: str
    risk_level: str
    recommendation: str
    matched_symptoms: list[str]


class RecommendationResult(BaseModel):
    planting_record_id: int
    highest_risk_level: str
    matches: list[RecommendationMatch]


class AlertRead(BaseModel):
    id: int
    planting_record_id: int
    symptom_record_id: int | None = None
    risk_level: str
    message: str
    is_read: bool

    model_config = {"from_attributes": True}

File: agrimonitor_frontend/src/types/recommendation.ts
export type RecommendationMatch = {
  disease_rule_id: number;
  disease_name: string;
  risk_level: "low" | "medium" | "high";
  recommendation: string;
  matched_symptoms: string[];
};

export type RecommendationResult = {
  planting_record_id: number;
  highest_risk_level: "none" | "low" | "medium" | "high";
  matches: RecommendationMatch[];
};

export type Alert = {
  id: number;
  planting_record_id: number;
  symptom_record_id: number | null;
  risk_level: string;
  message: string;
  is_read: boolean;
};